wallet/wallet-cli-lib/tests/basic.rs

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) 2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use tokio::sync::oneshot;
+use wallet_cli_lib::{
+    config::{Network, WalletCliArgs},
+    console::ConsoleContext,
+};
+
+#[tokio::test(flavor = "multi_thread", worker_threads = 1)]
+async fn basic_wallet_cli() {
+    // logging::init_logging::<std::path::PathBuf>(None);
+
+    let test_root = test_utils::test_root!("wallet-cli-tests").unwrap();
+    let test_dir = test_root.fresh_test_dir("basic_wallet_cli");
+
+    let commands_file = test_dir.as_ref().join("commands.txt");
```

Comment:
Why not use the vector input that was developed instead of depending on a file in a test? The whole point of that vector is to minimize file I/O to the point where our abstraction layer is so good that we basically are testing reading a file if we use a file.

Reply:
Minimizing IO is probably a bit pointless since we start the whole node here. Anyway, I switched to using a generic console implementation to abstract console IO.